e2e/util.go

```diff
@@ -75,7 +79,89 @@ sign_cert: {{.SignCert}}
 num_of_conn: 10
 client_per_conn: 10
 `
-	tmpl, err := template.New("test").Parse(Text)
+
+	var Text2 = `# Definition of nodes
+node1: &node1
+  addr: {{ .Addr }}
+  {{ if .Mtls }}
+  tls_ca_cert: {{.MtlsCrt}}
+  tls_ca_key: {{.MtlsKey}}
+  tls_ca_root: {{.MtlsCrt}}
+  {{ end }}
+node2: &node2
+  addr: {{ .Addr }}
+  {{ if .Mtls }}
+  tls_ca_cert: {{.MtlsCrt}}
+  tls_ca_key: {{.MtlsKey}}
+  tls_ca_root: {{.MtlsCrt}}
+  {{ end }}
+# Nodes to interact with
+endorsers:
+  - *node1
+  - *node2
+committers: 
+  - *node1
+  - *node2
+commitThreshold: {{ .CommitThreshold }}
+orderer: *node1
+channel: test-channel
+chaincode: test-chaincode
+mspid: Org1MSP
+private_key: {{.PrivSk}}
+sign_cert: {{.SignCert}}
+num_of_conn: 10
+client_per_conn: 10
+`
+
+	var Text3 = `# Definition of nodes
+node1: &node1
+  addr: {{ .Addr }}
+  {{ if .Mtls }}
+  tls_ca_cert: {{.MtlsCrt}}
+  tls_ca_key: {{.MtlsKey}}
+  tls_ca_root: {{.MtlsCrt}}
+  {{ end }}
+node2: &node2
+  addr: {{ .Addr }}
+  {{ if .Mtls }}
+  tls_ca_cert: {{.MtlsCrt}}
+  tls_ca_key: {{.MtlsKey}}
+  tls_ca_root: {{.MtlsCrt}}
+  {{ end }}
+node3: &node3
+  addr: {{ .Addr }}
+  {{ if .Mtls }}
+  tls_ca_cert: {{.MtlsCrt}}
+  tls_ca_key: {{.MtlsKey}}
+  tls_ca_root: {{.MtlsCrt}}
+  {{ end }}
+# Nodes to interact with
+endorsers:
+  - *node1
+  - *node2
+committers: 
+  - *node1
+  - *node2
+  - *node3
+commitThreshold: {{ .CommitThreshold }}
+orderer: *node1
+channel: test-channel
+chaincode: test-chaincode
+mspid: Org1MSP
+private_key: {{.PrivSk}}
+sign_cert: {{.SignCert}}
+num_of_conn: 10
+client_per_conn: 10
+`
+	var err error
+	var tmpl *template.Template
+	tmpl, err = template.New("test").Parse(Text)
+	if values.Peers == 2 {
+		tmpl, err = template.New("test").Parse(Text2)
+	}
+	if values.Peers == 3 {
+		tmpl, err = template.New("test").Parse(Text3)
+	}
```

Comment:
Go template is capable to handle array, so you don't need to write separate templates specifically for 2&3